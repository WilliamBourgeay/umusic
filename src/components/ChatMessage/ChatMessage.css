.ChatMessage {
  box-sizing: content-box;
  max-width: 60%;
}

.message-receive {
  float: left;
  background-color: rgb(102, 153, 204) !important;
  border-bottom-right-radius: 1.3em;
  border-bottom-left-radius: 1.3em;
  border-top-right-radius: 1.3em;
}

.message-send {
  float: right;
  background-color: '#f1f0f0';
  border-top-left-radius: 1.3em;
  border-bottom-left-radius: 1.3em;
  border-top-right-radius: 1.3em;
}

.color-text-receive {
  color: white !important;
}

.color-text-send {
  color: black !important;
}

.padding-12 {
  padding: 12px;
}

.padding-8 {
  padding: 8px;
}

.alias {
  color: rgba(0, 0, 0, 0.4);
  font-size: 12px;
  font-weight: normal;
  line-height: 1.1;
  margin-bottom: 1px;
  overflow: hidden;
  padding-left: 12px;
  text-overflow: ellipsis;
  white-space: nowrap;
}
