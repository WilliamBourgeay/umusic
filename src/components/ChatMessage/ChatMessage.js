// @flow

import React from 'react';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import './ChatMessage.css';

type Props = {
  message: string,
  isReceived: boolean,
  alias: string,
};

const getPosition = isReceived => (isReceived ? 'receive' : 'send');

const getAlias = (isReceived, alias) => (isReceived ? alias : '');

const ChatMessage = ({ message, isReceived, alias }: Props) => (
  <div className="padding-8">
    <Typography className="alias">{getAlias(isReceived, alias)}</Typography>
    <Paper
      className={`ChatMessage message-${getPosition(isReceived)}`}
      elevation={4}
      square
      aria-describedby="cond"
    >
      <Typography className={`padding-12 color-text-${getPosition(isReceived)}`} component="p">
        {message}
      </Typography>
    </Paper>
  </div>
);

export default ChatMessage;
