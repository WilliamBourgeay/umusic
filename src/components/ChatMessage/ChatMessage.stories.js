// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';
import ChatMessage from './ChatMessage';

const stories = storiesOf('Receive', module);

stories.add('Receive', () => (
  <div style={{ display: 'flex', flexDirection: 'column', width: '50%' }}>
    <ChatMessage alias="Okde" message="Je test" isReceived />
    <ChatMessage alias="Okde" message="Je test" isReceived={false} />
    <ChatMessage alias="Okde" message="Je test" isReceived={false} />
  </div>
));
