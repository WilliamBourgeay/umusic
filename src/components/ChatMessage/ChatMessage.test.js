import * as React from 'react';
import renderer from 'react-test-renderer';

import ChatMessage from './ChatMessage';

describe('ChatMessage', () => {
  it('should render a ChatMessage receive', () => {
    const tree = renderer.create(<ChatMessage isReceived message="mess" alias="hehe" />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render a ChatMessage not receive', () => {
    const tree = renderer.create(<ChatMessage isReceived={false} message="mess" alias="hehe" />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
