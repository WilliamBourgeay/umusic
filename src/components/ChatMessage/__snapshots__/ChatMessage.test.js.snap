// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage should render a ChatMessage not receive 1`] = `
<div
  className="padding-8"
>
  <p
    className="MuiTypography-root-1 MuiTypography-body1-10 alias"
  >
    
  </p>
  <div
    aria-describedby="cond"
    className="MuiPaper-root-36 MuiPaper-elevation4-42 ChatMessage message-send"
  >
    <p
      className="MuiTypography-root-1 MuiTypography-body1-10 padding-12 color-text-send"
    >
      mess
    </p>
  </div>
</div>
`;

exports[`ChatMessage should render a ChatMessage receive 1`] = `
<div
  className="padding-8"
>
  <p
    className="MuiTypography-root-1 MuiTypography-body1-10 alias"
  >
    hehe
  </p>
  <div
    aria-describedby="cond"
    className="MuiPaper-root-36 MuiPaper-elevation4-42 ChatMessage message-receive"
  >
    <p
      className="MuiTypography-root-1 MuiTypography-body1-10 padding-12 color-text-receive"
    >
      mess
    </p>
  </div>
</div>
`;
